pkgs/tools/misc/grub/default.nix

```diff
@@ -1,52 +1,61 @@
-{stdenv, fetchurl, autoreconfHook, texinfo, buggyBiosCDSupport ? true}:
+{stdenv, fetchurl, fetchpatch, autoreconfHook, texinfo, buggyBiosCDSupport ? true}:
 
+let
+  baseUrl = "https://sources.debian.org/data/main/g/grub/0.97-73/debian/patches";
```

Comment:
`sources.debian.org` is not version controlled, they tend to remove/change files  so we get 404s and broken hashes later.

Can you please try to find the patches in their gitlab repo: https://salsa.debian.org/grub-team/grub/tree/master/debian/patches  - they should all be there somewhere, although maybe not all int the `grub-team` project.

Reply:
You can change `baseUrl` to -> https://salsa.debian.org/grub-team/grub-legacy/tree/debian/0.97-73/debian/patches

The **corrected** hashes are the same.

Reply:
Hm! Though I hadn't verified elsewhere whether it was a safe location, I verified that other locations in Nixpkgs used `sources.debian.org`. You say "not version controlled", but I see the version in the URL, which I would assume [always match with a released version](https://sources.debian.org/data/main/g/grub/) (since I see those). Is it that even though they are versioned, they may change?

Reply:
We've had issues fetchpatch-ing from `sources.debian.org` before. The content is not guaranteed to be stable, they do sometimes change or remove files even under the same version in the URL.
It's still used in nixpkgs but let's try to avoid it in the future. So the gitlab repo is the best option if you can find the patches there.